main.py

```diff
@@ -10,15 +10,20 @@
 
 while True:
     dynamic_objects = []
-    reference_points = []
+    reference_point = None
     timestamp = time.time()
 
     # Processing frames
     for camera in cameras:
         camera.update_frame()
         dynamic_objects.extend(object_detection.get_dynamic_objects(camera))
-        reference_points.extend(vision_processing.ReferencePoint.from_apriltags(camera))
+        reference_point = vision_processing.ReferencePoint.from_apriltags(camera)
```

Comment:
The reference point is always the value from the _last_ camera? If that's the case I'd recommend taking this out of the loop and explicitly pulling the value from the last camera in the array. That would be a lot more intention revealing.